### PR TITLE
III-3853 Remove StringLiteral from UDB3\UiTID

### PR DESCRIPTION
**Pull request by grubolsch**
### Changed
Remove StringLiteral from UDB3\UiTID

---
Ticket: https://jira.uitdatabank.be/browse/III-3853

### CODE REVIEW

File: src/EventHandling/DelegateEventHandlingToSpecificMethodTrait.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ private function getHandleMethodName($event): ?string
 
         $expectedClass = $parameter->getClass();
 
-        if ($expectedClass->getName() !== get_class($event)) {
+        if ($expectedClass !== null && $expectedClass->getName() !== get_class($event)) {
```
Comment (LucWollants) [inline src/EventHandling/DelegateEventHandlingToSpecificMethodTrait.php:40]: If `$expectedClass` is `null` should this method not return `null`?

File: tests/Event/ReadModel/Permission/ProjectorTest.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ public function it_adds_permission_to_the_user_identified_by_the_createdby_eleme
             $payload
         );
 
-        $userId = new StringLiteral('123');
+        $userId = 'dcd1e123-0608-4824-afe3-99124feda64b';
 
         $this->userIdResolver->expects($this->once())
             ->method('resolveCreatedByToUserId')
```
Comment (JonasVHG) [inline tests/Event/ReadModel/Permission/ProjectorTest.php:74]: I guess this comes from a mergeConflict, but there is a StringLiteral in the next line, that could be removed.
Same with some other markResourceEditableByUser() in this file

File: src/EventHandling/DelegateEventHandlingToSpecificMethodTrait.php (hunk 1 above)
Comment (grubolsch) [reply src/EventHandling/DelegateEventHandlingToSpecificMethodTrait.php]: Changed it into an or

File: tests/Event/ReadModel/Permission/ProjectorTest.php (hunk 2 above)
Comment (grubolsch) [reply tests/Event/ReadModel/Permission/ProjectorTest.php:72]: Removed the other string literals

Comment (JonasVHG) [reply tests/Event/ReadModel/Permission/ProjectorTest.php:72]: My mistake, should have been more clear 🫣, they are also in `markResourceEditableByUser()` in `tests/Place/ReadModel/Permission/ProjectorTest.php`

Comment (grubolsch) [reply tests/Event/ReadModel/Permission/ProjectorTest.php:72]: Updated it

**Review by LucWollants — APPROVED** [review]

**Review by JonasVHG — APPROVED** [review]

**Pull request merged**